src/datalogging.cpp

```diff
@@ -1,108 +1,170 @@
+    if (!f) return false;
+    f.println(v);
+    f.close();
+    return true;
+}
+
+static bool appendCsvHeaderIfNew(const char* path) {
+    if (!sd.exists(path)) {
+        FsFile f = sd.open(path, O_WRITE | O_CREAT | O_TRUNC);
+        if (!f) return false;
+        f.println("UptimeMillis,Time,Latitude,Longitude,GPSAltitude,BaroAltitude,Pressure,Temperature,Humidity");
+        f.close();
+    }
+    return true;
+}
 
 void dataloggingInit() {
-    gnssQueue = xQueueCreate(4, sizeof(GnssData));
-    sensingQueue = xQueueCreate(4, sizeof(SensingData));
+    Serial.println("=============");
+    Serial.println(" Datalogging ");
+    Serial.println("=============");
+
+    pinMode(MISO, INPUT_PULLUP);
+    pinMode(SD_CS_PIN, OUTPUT);
+    digitalWrite(SD_CS_PIN, HIGH);
+    SPI.begin();
+    SPI.beginTransaction(SPISettings(1000000, MSBFIRST, SPI_MODE0));
+    for (int i = 0; i < 10; ++i) SPI.transfer(0xFF);
+    SPI.endTransaction();
+    Serial.println("SPI initialized");
+
+    SdSpiConfig cfg(SD_CS_PIN, DEDICATED_SPI, SD_SCK_MHZ(INIT_MHZ));
 
-    if (!fatfs.begin(&flash)) {
-        Serial.println("Failed to initialize filesystem!");
+    if (!sd.cardBegin(cfg)) {
+        Serial.print("\tcardBegin failed  sdErr=0x");
+        Serial.print(sd.sdErrorCode(), HEX);
+        Serial.print(" sdData=0x");
+        Serial.println(sd.sdErrorData(), HEX);
         return;
     }
+    Serial.println("\tCard initialized");
 
-    File file = fatfs.open(BOOTCOUNT_FILE, FILE_READ);
-    if (!file) {
-        bootcount = 1;
-        file = fatfs.open(BOOTCOUNT_FILE, FILE_WRITE);
-        if (file) {
-            file.print(bootcount);
-            file.close();
-        }
+    uint64_t cardSize = sd.card()->cardSize();
+    uint64_t cardSizeBytes = cardSize * 512;
+    Serial.print("\tCard size: ");
+    if (cardSizeBytes < (2ULL * 1024 * 1024)) {
+        Serial.print(cardSizeBytes / 1024);
+        Serial.println(" KB");
+    } else if (cardSizeBytes < (2ULL * 1024 * 1024 * 1024)) {
+        Serial.print(cardSizeBytes / (1024 * 1024));
+        Serial.println(" MB");
     } else {
-        bootcount = file.parseInt() + 1;
-        file.close();
-        file = fatfs.open(BOOTCOUNT_FILE, FILE_WRITE);
-        if (file) {
-            file.print(bootcount);
-            file.close();
-        }
+        Serial.print(cardSizeBytes / (1024 * 1024 * 1024));
+        Serial.println(" GB");
     }
 
-    char indexedLogFile[FILE_NAME_SIZE] = {0};
-    snprintf(indexedLogFile, FILE_NAME_SIZE, "/data/log_%lu.csv", (unsigned long)bootcount);
-    File log = fatfs.open(indexedLogFile, FILE_WRITE);
-    if (log) {
-        log.println("Time,Latitude,Longitude,Altitude,Humidity,Temperature,Pressure,AltitudeTime");
-        log.close();
+    if (!sd.begin(cfg)) {
+        Serial.print("\tFS mount failed sdErr=0x");
+        Serial.print(sd.sdErrorCode(), HEX);
+        Serial.print(" sdData=0x");
+        Serial.println(sd.sdErrorData(), HEX);
```

Comment:
is sd.beginning twice a thing bc of the lack of resistor or is the a copy paste thing